src/Functions/ReplaceRegexpImpl.h

```diff
@@ -176,25 +230,33 @@ struct ReplaceRegexpImpl
         ColumnString::Offset res_offset = 0;
         res_data.reserve(haystack_data.size());
         size_t haystack_size = haystack_offsets.size();
-        res_offsets.resize(haystack_size);
-
-        re2::RE2::Options regexp_options;
-        /// Don't write error messages to stderr.
-        regexp_options.set_log_errors(false);
+        res_offsets.resize_exact(haystack_size);
 
-        re2::RE2 searcher(needle, regexp_options);
+        std::shared_ptr<OptimizedRegularExpression> searcher;
+        try
+        {
+            searcher = std::make_shared<OptimizedRegularExpression>(Regexps::createRegexp<false, false, false>(needle));
```

Comment:
Why does searcher need to be a `shared_ptr`? Won't

```cpp
OptimizedRegularExpression searcher = Regexps::createRegexp<false, false, false>(needle);
```

do the job?

Reply:
I thought about it before. But removing shared_ptr will cause some inconvenients.
 
``` cpp
OptimizedRegularExpression searcher;
try
{
    searcher = Regexps::createRegexp<false, false, false>(needle); 
}
catch ...
``` 

Code as above will cause compiler error becuase `OptimizedRegularExpression` doesn't have default constructor.